File: 1813-maximum-erasure-value/maximum-erasure-value.java
class Solution {
    public int maximumUniqueSubarray(int[] nums) {
        int[] freq = new int[10001];
        int start = 0;
        int max = 0;
        int temp = 0;
        for(int i=0;i<nums.length;i++) {
            if( freq[nums[i]] == 1 ) {
                while(start < i && nums[start] != nums[i]) {
                    temp -=nums[start];
                    freq[nums[start]]--;
                    start++;
                    
                }
                temp -=nums[start];
                freq[nums[start]]--;
                start++;
            }
            freq[nums[i]]++;
            temp +=nums[i];
            max = Math.max(temp,max);
        }

        return max;
    }
}

File: 1813-maximum-erasure-value/maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        freq={}
        start=0
        temp=0
        maximum =0
        for i in range(len(nums)):
            while freq.get(nums[i],0):
                freq[nums[start]] -=1
                temp -=nums[start]
                start +=1
            temp +=nums[i]
            freq[nums[i]] =1
            maximum = max(maximum,temp)
        return maximum

        